### PR TITLE
feat(utils,modules-sdk): Remote query object to string array

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- transform remote query object back to string array
- get all fields and fields from given relations from a GraphQL schema

### CODE REVIEW

File: packages/utils/src/common/remote-query-object-to-string.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Transform a remote query object to a string array containing the chain of fields and relations
+ *
+ * @param fields
+ * @param parent
+ *
+ * @example
+ *
+ * const remoteQueryObject = = {
```
Comment (pKorsholm) [inline packages/utils/src/common/remote-query-object-to-string.ts]:
```suggestion
 * const remoteQueryObject = {
```

File: .changeset/calm-spiders-beg.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@medusajs/utils": patch
+"@medusajs/modules-sdk": patch
+---
+
+feat(utils): Remote query object to string array and fields and relations from GQL schema
```
Comment (pKorsholm) [inline .changeset/calm-spiders-beg.md]: This is a bit hard to understand, don't know if my suggestion is much better though 😂 
```suggestion
feat(utils): Convert remote query object to string array of fields and relations from GQL schema
```

**Review by pKorsholm — APPROVED** [review]
lgtm, just have a couple of nits 😄

**Comment by adrien2p** [comment]
@carlos-r-l-rodrigues should we merge this pr and the other util?

**Comment by carlos-r-l-rodrigues** [comment]
> @carlos-r-l-rodrigues should we merge this pr and the other util?

I think that makes sense to merge with this https://github.com/medusajs/medusa/pull/5213

**Comment by olivermrbl** [comment]
> I think that makes sense to merge with this https://github.com/medusajs/medusa/pull/5213

@adrien2p @carlos-r-l-rodrigues – sorry, I read this after having merged #5213. But if we resolve the conflicts, this should be ready to merge too

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**